spec/multisite/s3_store_spec.rb

```diff
@@ -41,4 +34,60 @@
       end
     end
   end
+
+  context 'removal from s3' do
+    before do
+      SiteSetting.s3_region = 'us-west-1'
+      SiteSetting.s3_upload_bucket = "s3-upload-bucket"
+      SiteSetting.s3_access_key_id = "s3-access-key-id"
+      SiteSetting.s3_secret_access_key = "s3-secret-access-key"
+      SiteSetting.enable_s3_uploads = true
+    end
+
+    describe "#remove_upload" do
+      let(:store) { FileStore::S3Store.new }
+      let(:client) { Aws::S3::Client.new(stub_responses: true) }
+      let(:resource) { Aws::S3::Resource.new(client: client) }
+      let(:s3_bucket) { resource.bucket("s3-upload-bucket") }
+      let(:s3_helper) { store.instance_variable_get(:@s3_helper) }
+
+      it "removes the file from s3 on multisite", type: :multisite do
+        conn.with_connection('default') do
+          store.expects(:get_depth_for).with(upload.id).returns(0)
+          s3_helper.expects(:s3_bucket).returns(s3_bucket).at_least_once
+          upload.update_attributes!(url: "//s3-upload-bucket.s3.dualstack.us-west-1.amazonaws.com/uploads/default/original/1X/#{upload.sha1}.png")
+          s3_object = stub
+
+          s3_bucket.expects(:object).with("uploads/tombstone/default/original/1X/#{upload.sha1}.png").returns(s3_object)
+          s3_object.expects(:copy_from).with(copy_source: "s3-upload-bucket/uploads/default/original/1X/#{upload.sha1}.png")
+          s3_bucket.expects(:object).with("uploads/default/original/1X/#{upload.sha1}.png").returns(s3_object)
+          s3_object.expects(:delete)
+
+          store.remove_upload(upload)
+        end
+      end
+
+      describe "when s3_upload_bucket includes folders path" do
+        before do
+          SiteSetting.s3_upload_bucket = "s3-upload-bucket/discourse-uploads"
+        end
+
+        it "removes the file from s3 on multisite", type: :multisite do
```

Comment:
We don't need `type: :multisite` here since it is already declared in at the start of the file